llvm/lib/Object/ArchiveWriter.cpp

```diff
@@ -807,6 +827,30 @@ computeMemberData(raw_ostream &StringTable, raw_ostream &SymNames,
     }
   }
 
+  if (SymMap) {
+    if (IsEC) {
+      SymMap->UseECMap = *IsEC;
```

Comment:
I've not attempted to thread this through, so it may well be, but is this case covered by tests?

Reply:
Yes, it's covered. `llvm-lib/empty.test` from this PR checks its corner cases. That's also what existing `llvm-lib` and import libraries in general use.